keps/sig-node/2535-ensure-secret-pulled-images/README.md

```diff
@@ -179,8 +179,7 @@ or expect preloaded images since boot.
 
 ## Design Details
 
-Kubelet will track, in memory, a hash map for the credentials that were successfully used to pull an image. The hash map
-will not be persisted to disk, in alpha. For alpha explicitly, we will not reuse or add other state manager concepts to kubelet.
+Kubelet will track, in memory, a hash map for the credentials that were successfully used to pull an image. It has been decided that the hash map will be persisted to disk, in alpha.
```

Comment:
> It has been decided that the hash map will be persisted to disk, in alpha.
Do you consider storing it in node status?

There are already something like 
```
    - names:
      - docker.io/calico/csi@sha256:028557c39d0a02dc47e2b9ae06fbdb638af8cce65b43540a13c7b1cf9c00ffcf
      - docker.io/calico/csi:v3.24.0
      sizeBytes: 9044267
```
To save if the secret is ensured, can we add a flag in this data structure directly?

https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/client-go/applyconfigurations/core/v1/containerimage.go#L23-L26

Reply:
This would be similar to `/var/lib/kubelet/cpu_manager_state` which saves the CPU manager policy. This is also a choice.

Reply:
for updating/secrets ... easy enough to "expire" hash map entries after a set number of hours/duration.. see comments in PR.

where to store the hash map, and in what form is tricky, but we have "root" to fall back on
and.. we can consider falling back to the key ring model for secret/injection overloading 
that is currently employed to also let the cloud provider provide a storage mechanism..

Reply:
I used `/var/lib/kubelet/image_manager_state` 
```
{
  "images": {
    "sha256:eb6cbbefef909d52f4b2b29f8972bbb6d86fc9dba6528e65aad4f119ce469f7a": {
      "authHash": {
        "115b8808c3e7f073": {
          "ensured": true,
          "dueDate": "2023-05-30T05:26:53.76740982+08:00"
        }
      },
      "name": "daocloud.io/daocloud/dce-registry-tool:3.0.8"
    }
  }
}
```
The data structure is like above.